planning/behavior_velocity_planner/src/scene_module/detection_area/scene.cpp

```diff
@@ -203,6 +203,67 @@ bool DetectionAreaModule::modifyPathVelocity(PathWithLaneId * path, StopReason *
   return true;
 }
 
+// calc smallest enclosing circle with average O(N) algorithm
+std::pair<lanelet::BasicPoint2d, double> calcSmallestEnclosingCircle(
+  const lanelet::ConstPolygon2d & poly)
+{
+  const double eps = 1e-5;
+  lanelet::BasicPoint2d center(0.0, 0.0);
+  double radius_squared = 0.0;
+
+  auto cross = [](lanelet::BasicPoint2d p1, lanelet::BasicPoint2d p2) -> double {
+    return p1.x() * p2.y() - p1.y() * p2.x();
+  };
+
+  auto make_circle_3 = [&](
```

Comment:
I think you had better to add reference links to show what A B C D means or you had better to add more concrete names to them.

Reply:
LGTM
for behavior anyway
![image](https://user-images.githubusercontent.com/65527974/217738016-c11ef91e-30bd-452c-85da-4af6a18c687b.png)

Reply:
Thank you for your review.
I just added a reference for calculating the circumcenter vector.